db/context.go

```diff
@@ -53,6 +53,11 @@ func ensureContextTxn(ctx context.Context, db transactionDB, readOnly bool) (con
 	return SetContextTxn(ctx, txn), txn, nil
 }
 
+// mustGetContextTxn returns the transaction from the context or panics.
+func mustGetContextTxn(ctx context.Context) datastore.Txn {
+	return ctx.Value(txnContextKey{}).(datastore.Txn)
```

Comment:
todo: This is a user input, and I think an error if not found would be much better than a panic - unless it is always guaranteed to exist at the point that this function is called, in which case please document this in this function.

Reply:
I've updated the docs to better reflect our usage pattern. It should be guaranteed to exist on any private function with `db`.

Reply:
Nice - new doc changes look good, thanks Keenan :)